data/collector.go

```diff
@@ -62,12 +63,19 @@ func (c *collector) Close() {
 	if err := c.target.Flush(); err != nil {
 		c.logger.Errorw("failed to flush capture queue", "error", err)
 	}
-	if err := c.logger.Sync(); err != nil {
-		c.logger.Errorw("failed to sync logger", "error", err)
-	}
+	//nolint:errcheck
+	_ = c.logger.Sync()
```

Comment:
If the logger is failing, we can't log. Sync is also not supported by all loggers (ones that write to stdout/stderr will give [this](https://stackoverflow.com/questions/1605195/inappropriate-ioctl-for-device) error on linux machines, for example). So this does a best effort attempt at syncing, and moves on if it fails.